src/ckb_chain/actor.rs

```diff
@@ -214,29 +217,361 @@ impl CkbChainState {
-    type State = ();
-    type Arguments = ();
+mod test_utils {
+    use std::collections::HashMap;
 
-    async fn pre_start(
-        &self,
-        _: ActorRef<Self::Msg>,
-        _: Self::Arguments,
-    ) -> Result<Self::State, ActorProcessingErr> {
-        Ok(())
+    use ckb_types::{
+        core::TransactionView,
+        packed::CellOutput,
+        prelude::{Builder, Entity, Pack, PackVec, Unpack},
+    };
+
+    use crate::ckb_chain::TraceTxRequest;
+
+    use super::super::contracts::MockContext;
+    use super::CkbChainMessage;
+
+    use ckb_types::packed::Byte32;
+    use log::{debug, error};
+    use ractor::{call_t, Actor, ActorProcessingErr, ActorRef};
+
+    pub struct MockChainActorState {
+        ctx: MockContext,
+        committed_tx_status: HashMap<Byte32, ckb_jsonrpc_types::Status>,
     }
 
-    async fn handle(
-        &self,
-        _: ActorRef<Self::Msg>,
-        _: Self::Msg,
-        _: &mut Self::State,
-    ) -> Result<(), ActorProcessingErr> {
-        Ok(())
+    impl MockChainActorState {
+        pub fn new() -> Self {
+            Self {
+                ctx: MockContext::new(),
+                committed_tx_status: HashMap::new(),
+            }
+        }
+    }
+
+    pub struct MockChainActor {}
+
+    impl MockChainActor {
+        pub fn new() -> Self {
+            Self {}
+        }
+    }
+
+    #[ractor::async_trait]
+    impl Actor for MockChainActor {
+        type Msg = CkbChainMessage;
+        type State = MockChainActorState;
+        type Arguments = ();
+
+        async fn pre_start(
+            &self,
+            _: ActorRef<Self::Msg>,
+            _: Self::Arguments,
+        ) -> Result<Self::State, ActorProcessingErr> {
+            Ok(Self::State::new())
+        }
+
+        async fn handle(
+            &self,
+            myself: ActorRef<Self::Msg>,
+            message: Self::Msg,
+            state: &mut Self::State,
+        ) -> Result<(), ActorProcessingErr> {
+            use CkbChainMessage::*;
+            match message {
+                Fund(tx, request, reply_port) => {
+                    let mut fulfilled_tx = tx.clone();
+                    let outputs = fulfilled_tx
+                        .as_ref()
+                        .map(|x| x.outputs())
+                        .unwrap_or_default();
+                    let outputs = match outputs.get(0) {
+                        Some(output) => {
+                            if output.lock() != request.script {
+                                error!(
+                                        "funding request script ({:?}) does not match the first output lock script ({:?})", request.script, output.lock()
+                                    );
+                                return Ok(());
+                            }
+                            let current_capacity: u64 = output.capacity().unpack();
+                            let capacity = request.local_amount + current_capacity;
```

Comment:
emm, we need to care about UDT on this part.

Reply:
Yeah. We should also take care of UDT here, which would depend on your work. So, for now, we'd better leave this as a known issue.